### PR TITLE
Installation doc

### PR DESCRIPTION
**Pull request by himanshu-02**
Removed previous windows_installation folder which was deprecated
Updated the shell scripts for windows and macos
New installation doc

### CODE REVIEW

**Comment by gautam-dev-maker** [comment]
This PR looks good to me, @SAtacker could you check it once.

File: wall_e_install_mac.sh [hunk 1]
```diff
@@ -56,8 +56,16 @@ fi
...
+    cd ~/esp/esp-idf
+    git rm -rf .
+    git clean -fxd
+    rm -rf .git
+    cd ..
+    rm -r esp-idf
+    rm -rf .git
+    cd ..
     rm -R esp
     mkdir esp
```
Comment (SAtacker) [inline wall_e_install_mac.sh:70]: Why not `rm -rf "$HOME"/esp` instead of this?

File: wall_e_install.sh [hunk 2]
```diff
@@ -7,9 +7,9 @@ else
     echo "Installing ESP IDF"
     sudo apt-get update
     sudo apt-get upgrade
-    sudo apt-get install git wget libncurses-dev flex bison gperf python python-pip python-setuptools python-serial python-click python-cryptography python-future python-pyparsing python-pyelftools cmake ninja-build ccache
-    cd ~
-    mkdir esp && cd esp
+    sudo apt-get install git wget libncurses-dev flex bison gperf python python3-pip python-setuptools python-serial python-click python-cryptography python-future python-pyparsing python-pyelftools cmake ninja-build ccache
+    mkdir esp
+    cd ~/esp
```
Comment (SAtacker) [inline wall_e_install.sh:12]: `mkdir -p "$HOME/"esp && pushd "$HOME/"esp || echo "Error" && exit 1`

Comment (SAtacker) [reply wall_e_install.sh:12]: This needs more work, create a separate PR for it.

**Review by SAtacker — CHANGES_REQUESTED** [review]
We should make separate PRs for docs and installation scripts.

**Pull request closed without merging**